### ISSUE TITLE
Pen stuck on first class in list after trying to switch

### ISSUE THREAD

**Issue by xandra22** [dbuscombe-usgs/dash_doodler#32]
Hello,

I've been having some trouble getting the latest version of doodler to work on my machine. I'm largely following this guide, https://doodleverse.github.io/dash_doodler/docs/tutorial-basics/deploy-local/, to install it (although I have to install a few more packages in anaconda before the environment works fully and doodler can be launched in my browser). 

When I get to the doodler page, it looks normal and loads my assets, but when I go back to doodle on an image it will not switch from the first color/class in the list - in this case, "water". Here's a screenshot of some doodles when I tried to switch it to the "cloud" and "nodata" classes in my class file:

![screenshotOfDoodler_samePenColor](https://user-images.githubusercontent.com/98843496/171698040-7f24df9f-142b-4de9-9982-c0dff7799690.PNG)

I noticed there were some errors being thrown up in the terminal when I tried switching the pen:

![terminal_error_uponSwitchingClasses](https://user-images.githubusercontent.com/98843496/171698101-9ae79884-e4b1-417e-a7de-66d22fb3b1cf.PNG)

Is it on my end? Maybe there's something wrong with my environment? I'm using a windows machine: 

![image](https://user-images.githubusercontent.com/98843496/171698646-73769613-8ef4-4127-be8b-90179cc26829.png)

and I tried opening doodler in both Microsoft Edge (which worked for me before with previous versions of doodler) and Internet Explorer. Here's my class file and an example image:

[classes.txt](https://github.com/Doodleverse/dash_doodler/files/8826509/classes.txt)
![tiled_Job712109_2019_NGS_NaturalColorImagery_001_001_16000_7000](https://user-images.githubusercontent.com/98843496/171698464-d7bae29e-d959-41bf-b3ce-4b50e104cbe4.jpg)

**Comment by xandra22** [comment]
Update: I tried installing the environment via pip and it's still presenting the same issue of the pen not switching classes. Same error in terminal, as well, with line 595 in app.py as shown in prior screenshot.

**Comment by ddbuscombe** [comment]
Hi @xandra22, thanks for the update. I could not reproduce the issue (see below) using Firefox or Chrome (I dont have Edge or IE on this machine). I doodled cloud first, then  water, then nodata (although you should know that if the image is all zeros, Doodler will automatically assign those pixels with the label 0)

![Screenshot from 2022-06-14 09-22-47](https://user-images.githubusercontent.com/3596509/173627681-e612616c-2bfe-4f18-92c1-53747c7e58ff.png)

I dont know why you couldnt change pens. It could be a browser issue. It doesnt seem likely at all it would be an issue with the conda environment, but it is possible ....

**Comment by ddbuscombe** [comment]
The error message you got is possibly cryptic. It refers to not being able to read or parse the filename. No spaces in filenames? All images inside the assets folder, not subfolders?

**Comment by ddbuscombe** [comment]
@xandra22 you may be interested in trying out an experimental branch I just pushed to https://github.com/dbuscombe-usgs/dash_doodler. If you use it and see improvements, please report here thanks!

**Comment by xandra22** [comment]
The experimental branch worked for me! But first I had to uninstall anaconda and re-install it. Something was off on my end I think. After I re-installed it, I installed the environment with both mamba and conda and they both worked (mamba took two tries? but it was a lot faster to install just so people are aware - unsure why the first time i used mamba to install it the sample images worked and not my own, but upon a second try they both worked.). I used the updated yml file and it worked beautifully. I'll try doodling 10 or so of our images and report back on how well the new version performs on the github page Dan linked above.

and side note regarding the "cryptic comment" above from Dan: there were no spaces in filenames and the images were all inside the assets folder and not subfolders, so i'm still not sure what was happening. i saw that Cameron was able to get it working with the clean version of the yml file, but with the new doodler version i won't bother trying that fix unless i run into the issue again.
